### PR TITLE
Add a check_dimensions_match function

### LINKED ISSUE #499: Add a simple "check dimensions match" method
We currently have [lots of places](https://github.com/allenai/allennlp/blob/master/allennlp/models/reading_comprehension/bidaf.py#L100-L118) in our model init methods that take 5 lines to do a simple dimension check, and it's onerous to add those checks, so they don't get added a lot of the time.  We should have a simple function to call that checks that two dimensions match, and throws a `ConfigurationError` if they don't.

### PR DESCRIPTION
**Pull request by matt-gardner**
Fixes #499.

### CODE REVIEW

File: allennlp/common/checks.py [hunk 1]
```diff
@@ -24,3 +24,12 @@ def __str__(self):
 def log_pytorch_version_info():
     import torch
     logger.info("Pytorch version: %s", torch.__version__)
+
+
+def check_dimensions_match(dimension_1: int,
+                           dimension_2: int,
+                           dim_1_name: str,
+                           dim_2_name: str) -> None:
```
Comment (joelgrus) [inline allennlp/common/checks.py]: do you want to add an optional `message` field, so that we can include longer explanations when appropriate? (I see that this change currently eliminates some of them)

Comment (matt-gardner) [reply allennlp/common/checks.py]: I think the only place where there is information lost is in saying that the SRL model uses a binary verb indicator, which is actually incorrect (it was originally binary, it's now an embedding).  You still get the gist of (the correct part of) that information in the `"text embedding dim + verb indicator embedding dim"` message, so I'm not sure we're losing anything important.  If there's more to say, it should probably be said in the model docstring.
